firebase_connector/models/firebase_registration.py

```diff
@@ -0,0 +1,91 @@
+# -*- coding: utf-8 -*-
+##############################################################################
+#
+#    Copyright (C) 2019 Compassion CH (http://www.compassion.ch)
+#    @author: Nicolas Badoux <n.badoux@hotmail.com>
+#
+#    The licence is in the file __manifest__.py
+#
+##############################################################################
+import logging
+import firebase_admin
+from firebase_admin import credentials
+from firebase_admin import messaging
+
+from odoo import api, models, fields
+from odoo.tools import config
+
+
+_logger = logging.getLogger(__name__)
+
+
+class FirebaseRegistration(models.Model):
+    """
+    The represent a device that we can send push notification to. It is
+    characterized by the Firebase token and an optional partner.
+    It provides function to send messages to the device.
+    """
+
+    _name = 'firebase.registration'
+    _description = 'Device registered with Firebase Cloud Messaging'
+
+    registration_id = fields.Char(required=True,
+                                  string="Firebase Registration ID")
+    partner_id = fields.Many2one('res.partner', string="Partner")
+    partner_name = fields.Char(related='partner_id.name', readonly=True)
+    firebase_credentials =\
+        credentials.Certificate(config.get('firebase_api_key'))
+    firebase_app = firebase_admin.initialize_app(
```

Comment:
Are you sure you want to do this at the class level? Plus it looks like this variable is never used?

Reply:
It's used internally by firebase.